source/adapters/opencl/context.cpp

```diff
@@ -14,38 +14,22 @@
 #include <set>
 #include <unordered_map>
 
-ur_result_t cl_adapter::getDevicesFromContext(
-    ur_context_handle_t hContext,
-    std::unique_ptr<std::vector<cl_device_id>> &DevicesInCtx) {
-
-  cl_uint DeviceCount;
-  CL_RETURN_ON_FAILURE(clGetContextInfo(cl_adapter::cast<cl_context>(hContext),
-                                        CL_CONTEXT_NUM_DEVICES, sizeof(cl_uint),
-                                        &DeviceCount, nullptr));
-
-  if (DeviceCount < 1) {
-    return UR_RESULT_ERROR_INVALID_CONTEXT;
-  }
-
-  DevicesInCtx = std::make_unique<std::vector<cl_device_id>>(DeviceCount);
-
-  CL_RETURN_ON_FAILURE(clGetContextInfo(
-      cl_adapter::cast<cl_context>(hContext), CL_CONTEXT_DEVICES,
-      DeviceCount * sizeof(cl_device_id), (*DevicesInCtx).data(), nullptr));
-
-  return UR_RESULT_SUCCESS;
-}
-
 UR_APIEXPORT ur_result_t UR_APICALL urContextCreate(
     uint32_t DeviceCount, const ur_device_handle_t *phDevices,
     const ur_context_properties_t *, ur_context_handle_t *phContext) {
 
   cl_int Ret;
-  *phContext = cl_adapter::cast<ur_context_handle_t>(
-      clCreateContext(nullptr, cl_adapter::cast<cl_uint>(DeviceCount),
-                      cl_adapter::cast<const cl_device_id *>(phDevices),
-                      nullptr, nullptr, cl_adapter::cast<cl_int *>(&Ret)));
+  std::vector<cl_device_id> CLDevices(DeviceCount);
+  for (size_t i = 0; i < DeviceCount; i++) {
+    CLDevices[i] = phDevices[i]->get();
+  }
 
+  cl_context Ctx = clCreateContext(
+      nullptr, cl_adapter::cast<cl_uint>(DeviceCount), CLDevices.data(),
+      nullptr, nullptr, cl_adapter::cast<cl_int *>(&Ret));
```

Comment:
If `Ret != CL_SUCCESS` then we should return `Ret` not allocate an invalid `ur_context_handle_t`.

Reply:
Fixed that and tried to move to other places to make sure everywhere have this check.